lib/cost_report.rb

```diff
@@ -1,15 +1,89 @@
 # frozen_string_literal: true
 
 require "ht_item_overlap"
+require "utils/waypoint"
+require "services"
 
 # Generates reports based on h_share
 class CostReport
-  attr_accessor :organization, :freq_table
+  attr_accessor :organization, :logger, :maxlines, :target_cost
 
-  def initialize(org = nil)
+  def initialize(org = nil, cost: ENV["target_cost"], lines: 1_000_000, logger: Services.logger)
     @organization = org
-    # { org => { 1 org : count, 2 org : count }
-    @freq_table = Hash.new {|hash, key| hash[key] = Hash.new(0) }
+    @target_cost = Float(cost)
+    @maxlines = lines
+    @logger = logger
+    # Member Hash of a format hash of a member count hash
+    # { org => { ser : { 1 org : count, 2 org : count }, mpm : {...
+    @freq_table = Hash.new do |hash, member|
+      hash[member] = Hash.new {|fmt_hash, fmt| fmt_hash[fmt] = Hash.new(0) }
+    end
+  end
+
+  def num_volumes
```

Comment:
Just raising here the idea that at some point we should extract all these mongo queries into methods that can be reused throughout the codebase. Mongoid queries support composition, so they'll come in handy.